pkg/registry/apis/provisioning/controller/finalizers.go

```diff
@@ -88,7 +89,7 @@ func (f *finalizer) processExistingItems(
 	cb func(client dynamic.ResourceInterface, item *provisioning.ResourceListItem) error,
 ) error {
 	logger := logging.FromContext(ctx)
-	client, _, err := f.client.New(repo.Namespace)
+	clients, err := resources.NewResourceClients(ctx, f.configProvider, repo.Namespace)
```

Comment:
I think that keeping the factory will helps us with testing. Can you keep it there and simply change how the clients are built?